fastplotlib/layouts.py

```diff
@@ -7,6 +7,12 @@
 from wgpu.gui.auto import WgpuCanvas
 
 
+class GridPlotIndexer:
+    def __init__(self, position):
+        self.name: str = None
```

Comment:
We need a way to make sure that each `GridPlotIndexer` instance within a `GridPlot` is unique, so we can add the attribute `parent` which is the parent `GridPlot` instance.

We should make `self.name` a property, so when the `name` is being set it checks to make sure it's not used by another `GridPlot` instance